### DISCUSSION TITLE
Renovate does not pick up new sbt plugin updates by default

### DISCUSSION THREAD

**Discussion by george-wilson-rea** [renovatebot/renovate#29755] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Repo 1: https://github.com/george-wilson-rea/renovate-scala-plugins
Repo 2: https://github.com/george-wilson-rea/renovate-scala-plugins-fixed

## Problem

In the first repo, Renovate does not update the `sbt-scoverage` dependency. I expect that this dependency should be updated.
Both repos set `sbt-scoverage` to version `2.0.11` and a newer version `2.0.12` is [available here](https://repo1.maven.org/maven2/org/scoverage/sbt-scoverage_2.12_1.0/2.0.12/).
In Repo 2, I have overridden Renovate's sbt-plugin `registryUrls` and I get a plugin update as expected.

## Config

Repo 1 uses the [default renovate configuration](https://github.com/george-wilson-rea/renovate-scala-plugins/blob/main/renovate.json#L4).
Repo 2 uses a [different config](https://github.com/george-wilson-rea/renovate-scala-plugins-fixed/blob/main/renovate.json#L10-L11) which lists the `https://repo1.maven.org/maven2/` URL first.

## Behaviour

In the first case, Renovate is only checking the `https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases` URL for the `sbt-scoverage` plugin. Unfortunately that repository only has older plugin releases (2021 and older). See [this post](https://www.scala-sbt.org/1.x/docs/Bintray-For-Plugins.html) for context. Renovate needs to look at the newer SBT plugin repo, which is `https://repo1.maven.org/maven2/`. It seems like renovate is not looking at this location for `sbt-scoverage`.

In Repo 2, I've configured the newer SBT plugin repo to be first in the list, so I get the expected dependency update.
I've opened #29648 to update the defaults accordingly.

### Logs (if relevant)

<details><summary>Logs</summary>

```

      {
        "deps": [
          {
            "datasource": "sbt-plugin",
            "depName": "org.scoverage:sbt-scoverage",
            "packageName": "org.scoverage:sbt-scoverage",
            "currentValue": "2.0.11",
            "depType": "plugin",
            "registryUrls": [
              "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases",
              "https://repo1.maven.org/maven2"
            ],
            "updates": [],
            "versioning": "ivy",
            "warnings": [],
            "registryUrl": "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases",
            "dependencyUrl": "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scoverage/sbt-scoverage",
            "currentVersion": "2.0.11",
            "fixedVersion": "2.0.11"
          },
          {
            "datasource": "sbt-plugin",
            "depName": "org.wartremover:sbt-wartremover",
            "packageName": "org.wartremover:sbt-wartremover",
            "currentValue": "3.1.5",
            "depType": "plugin",
            "registryUrls": [
              "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases",
              "https://repo1.maven.org/maven2"
            ],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "3.1.6",
                "newValue": "3.1.6",
                "newMajor": 3,
                "newMinor": 1,
                "newPatch": 6,
                "updateType": "patch",
                "branchName": "renovate/org.wartremover-sbt-wartremover-3.x"
              }
            ],
            "versioning": "ivy",
            "warnings": [],
            "sourceUrl": "https://github.com/wartremover/wartremover",
            "registryUrl": "https://repo1.maven.org/maven2",
            "dependencyUrl": "https://repo1.maven.org/maven2/org/wartremover/sbt-wartremover",
            "currentVersion": "3.1.5",
            "isSingleVersion": true,
            "fixedVersion": "3.1.5"
          }
        ],
        "packageFile": "project/plugins.sbt"
      }

<snip>

    "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scoverage/sbt-scoverage/": {
      "GET": {
        "200": 1
      }
    },
    "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scoverage/sbt-scoverage/scala_2.10/": {
      "GET": {
        "200": 1
      }
    },
    "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scoverage/sbt-scoverage/scala_2.10/sbt_0.13/": {
      "GET": {
        "200": 1
      }
    },

```

</details>

**Comment by rarkins** [comment]
@zharinov could you take a look?

**Reply by george-wilson-rea** [reply]
I just fixed my second replication repo so I've edited my question accordingly.

**Comment by MPV** [comment]
Let's close this one now that the fix is merged?
- https://github.com/renovatebot/renovate/pull/29648
